src/main/java/org/dice_research/opal/civet/metrics/DateFormatMetric.java

```diff
@@ -0,0 +1,119 @@
+
+package org.dice_research.opal.civet.metrics;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.SimpleSelector;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.jena.vocabulary.DCTerms;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+/**
+ * For the Date Format metric, the following will be checked: Check, if the date
+ * format is according to the W3c standards or not.
+ * 
+ * According to DCAT https://www.w3.org/TR/vocab-dcat-2/#bib-datetime the
+ * correct formats are here: https://www.w3.org/TR/NOTE-datetime. All formats
+ * have to be checked.
+ * 
+ * Here the implemented code checks the turtle file whether the given date
+ * format is according to W3c standard or not, if it is of the standard format
+ * it gives 5 stars else 0 star.
+ * 
+ * Finally, an average of the following 4 cases has to be computed for the final
+ * score: dataset: issued + modified, distributions: issued + modified
+ * 
+ * @author Aamir Mohammed
+ */
+public class DateFormatMetric implements Metric {
+
+	private static final String DESCRIPTION = "If the date format is in the correct format then give 5 stars else 0 star"
+			+ "Finally, an average of the following 4 cases has to be computed for the finalscore: dataset: issued + modified, distributions: issued + modified";
+
+	public static int checkDateFormat(String issued) {
+		/*
+		 * Following are the correct date formats according to "w3.org". Here, we are
+		 * checking different date formats using regular expression.
+		 * 
+		 */
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}")) {
+			// "YYYY-MM-DD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}")) {
+			// "YYYY"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}")) {
+			// "YYYY-MM"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}:\\d{2}[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mm:ssTZD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mmTZD"
+			return 5;
+		}
+		if (issued.matches("\\d{4}-\\d{2}-\\d{2}[T]\\d{2}:\\d{2}:\\d{2}[.]\\d+[+]\\d{2}:\\d{2}")) {
+			// "YYYY-MM-DDThh:mm:ss.sTZD"
+			return 5;
+		}
+		return 0;
+	}
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		int result = 0;
+		int countIssued = 0;
+		int countModified = 0;
+		Resource dataSet = model.createResource(datasetUri);
+		if (dataSet.hasProperty(DCTerms.issued)
+				&& !(dataSet.getProperty(DCTerms.issued).getObject().toString().isEmpty())) {
```

Comment:
Repitition sorry :D :D 

This branch condition will not check the following case.
dct:issued ""; or when
dct:modified "";

And the dataset will get full 5-star ratings. So how to resolve this problem. The answer is to increment "countIssued" outside this branch condition. This variable must be very specifically incremented only when there is a relevant property(issued /modified) irrespective of the object.

In your test case, make dct:issued "" for the dataset, http://projekt-opal.de/dataset/_mcloudde_standortederdauerzhlstellenradverkehrindsseldorf and run the JunitTest. You will still get 5 stars. I hope this helps.

Reply:
I got now, actually, this issue was handled before using else statement, but according to the feedback, I removed else, but yeah I will look that thing now.

Reply:
I have resolved this one.

Reply:
Yes looks like resolved. Good job !!